File: core/universal/infra_engine.py
from __future__ import annotations

def parse_infra(text: str):
    src = (text or '').lower()
    infra = []
    for k in ['terraform', 'kubernetes', 'helm', 'dockerfile', 'compose']:
        if k in src:
            infra.append(k)
    return {"infra_components": sorted(set(infra))}

File: src/universal/infraEngine.ts
/**
 * Converted from Python: core/universal/infra_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function parseInfra(text: any): any {
  var src: any = String(py.or2(text, () => (""))).toLowerCase();
  var infra: any[] = [];
  var k: any;
  for (k of py.iter(["terraform", "kubernetes", "helm", "dockerfile", "compose"])) {
    if (py.contains(src, k)) {
      py.listAppend(infra, k);
    }
  }
  return {"infra_components": py.sorted(py.toSet(infra))};
}
